lib/figgy/root.rb

```diff
@@ -0,0 +1,62 @@
+class Figgy
+  class Root
+    @handlers = []
+
+    class << self
+      attr_accessor :handlers
+    end
+
+    def initialize(source, path)
+      @source = source
+      @path = path
+    end
+  end
+
+  class VaultRoot < Root
+    def initialize(source, path)
+      super
+    end
+
+    def fetch(overlay, config_key)
+      secret = @source.logical.read(File.join(@path, overlay.to_s, config_key))
+      secret ? [JSON.parse(secret.data.to_json)] : nil
+    end
```

Comment:
I would assume so, but just wanna make sure, this is cached right?

Reply:
`Store` should [cache the values of config keys](https://github.com/enova/figgy/blob/c2444c56d1576f786713abc8fa31b2de59e7cd51/lib/figgy/store.rb#L20) once fetched for the first time. 

Though that does remind me that [all_key_names ISN'T cached](https://github.com/sshao/figgy/blob/d1490ab828110625a64a57138d8a28d69d9e6e09/lib/figgy/finder.rb#L20) so right now every time a new key is fetched, that could potentially lead to another [vault list call](https://github.com/sshao/figgy/blob/d1490ab828110625a64a57138d8a28d69d9e6e09/lib/figgy/root.rb#L26).

I was going to leave it as-is for now because as keys are fetched and cached, list calls will also cease, since it's only called when fetching a new key with `Finder#load`